rosetta-source/src/main/rosetta/product-template-type.rosetta

```diff
@@ -85,42 +85,45 @@ type EconomicTerms: <" This class represents the full set of price-forming featu
         if payout -> assetPayout is absent
         then // Not Security Lending/Repo so must be Interest Rate Swap, which must set an exercise option type
             if terminationProvision -> extendibleProvision exists
-            then (terminationProvision -> extendibleProvision -> americanExercise exists
-                        and terminationProvision -> extendibleProvision -> bermudaExercise is absent
-                        and terminationProvision -> extendibleProvision -> europeanExercise is absent)
-                    or (terminationProvision -> extendibleProvision -> americanExercise is absent
-                        and terminationProvision -> extendibleProvision -> bermudaExercise exists
-                        and terminationProvision -> extendibleProvision -> europeanExercise is absent)
-                    or (terminationProvision -> extendibleProvision -> americanExercise is absent
-                        and terminationProvision -> extendibleProvision -> bermudaExercise is absent
-                        and terminationProvision -> extendibleProvision -> europeanExercise exists)
+            then (terminationProvision -> extendibleProvision -> exerciseTerms -> style = OptionExerciseStyleEnum -> American 
```

Comment:
The next 9 statements are redundant since `style` is an enum - That's the benefit of handling the switch through a single, common type plus an enum.

Reply:
Noted, thanks. I've removed these 9 statements and got rid of the condition altogether since both `exerciseTerms` and `style` are mandatory within `extendibleProvision`, so one of the 3 styles of exercise will always be present.